samples/hello-cube.html

```diff
@@ -25,144 +25,167 @@
     modelViewProjectionMatrix: mat4x4<f32>;
 };
 
-[[group(${bindGroupIndex}), binding(${transformBindingNum})]] var<uniform> uniforms;
+[[group(${bindGroupIndex}), binding(${transformBindingNum})]] var<uniform> uniforms: Uniforms;
+
+let pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
+    vec2<f32>(0.0, 0.5),
+    vec2<f32>(-0.5, -0.5),
+    vec2<f32>(0.5, -0.5));
 
 [[stage(vertex)]]
-fn main(
-    [[location(${positionAttributeNum})]] position: vec4<f32>
-    [[location(${colorAttributeNum})]] color: vec4<f32>,
+fn vertex_main(
+    [[builtin(vertex_index)]] VertexIndex : u32,
+    [[location(${positionAttributeNum})]] position: vec4<f32>,
+    [[location(${colorAttributeNum})]] color: vec4<f32>
 ) -> FragmentData {
-    FragmentData out;
-    out.position = mul(uniforms.modelViewProjectionMatrix[0], position);
-    out.color = color;
-    return out;
+    var outData: FragmentData;
+    outData.position = uniforms.modelViewProjectionMatrix * position;
+    outData.color = color;
+    return outData;
 }
 
 [[stage(fragment)]]
-fn main(data: FragmentData) -> [[location(0)]] vec4<f32> {
+fn fragment_main(data: FragmentData) -> [[location(0)]] vec4<f32> {
     return data.color;
 }
 `;
 
-let device, swapChain, verticesBuffer, bindGroupLayout, pipeline, renderPassDescriptor;
-let projectionMatrix = mat4.create();
-let mappedGroups = [];
+let device, swapChain, verticesBuffer, renderPipeline, renderPassDescriptor;
+let transformBuffer, bindGroup;
+const projectionMatrix = mat4.create();
```

Comment:
why are we mixing const and let?

Reply:
`const` for the variables that don't need to change later, `let` for those that do